meteor/sofie-server/model/entities/test/entity-mock-factory.ts

```diff
@@ -0,0 +1,197 @@
+import { Segment, SegmentInterface } from '../segment'
+import { Part, PartInterface } from '../part'
+import { Piece, PieceInterface } from '../piece'
+import { PieceType } from '../../enums/piece-type'
+import { PieceLifespan } from '../../enums/piece-lifespan'
+import { Rundown } from '../rundown'
+import { anything, deepEqual, instance, mock, when } from 'ts-mockito'
+import { TransitionType } from '../../enums/transition-type'
+import { PartTimings } from '../../value-objects/part-timings'
+
+export class EntityMockFactory {
+	public static createActiveRundown(
+		activeRundownProperties: Partial<{
+			activePart?: Part
+			nextPart?: Part
+			previousPart?: Part
+			activeSegment?: Segment
+			nextSegment?: Segment
+			infinitePieces?: Piece[]
+		}> = {}
```

Comment:
Since all the attributes of the object inside the `Partial<>` is already optional, is the `Partial<>` even needed?

Reply:
You are right.
It seems in my eager to apply Anders suggestion I took it a step too far without thinking.
Good catch :) 

It's been updated